src/main/java/io/github/mirromutth/r2dbc/mysql/ExceptionFactory.java

```diff
@@ -57,30 +63,50 @@ static R2dbcException createException(ErrorMessage message, @Nullable String sql
 
     @Nullable
     private static R2dbcException mappingErrorCode(String errorMessage, @Nullable String sqlState, int errorCode, @Nullable String sql) {
-        // TODO: support more error codes
+        // Should keep looking more error codes
+        // Try use binary search or primitive-int-to-primitive-int hashed map?
```

Comment:
Using a `switch` has O(1) time complexity whereas a `Map` looks nicer but requires lookups. That was the driving rationale in r2dbc-mssql.